drivers/clock_control/clock_control_wch_rcc.c

```diff
@@ -40,6 +40,10 @@
 #define WCH_RCC_SRC_IS_HSE 1
 #endif
 
+#if defined(CONFIG_DT_HAS_WCH_CH32V20X_30X_PLL_CLOCK_ENABLED)
+#define CH32V20X_30X_PLL_MUL DT_PROP(DT_NODELABEL(pll), mul)
+#endif
```

Comment:
There is an opportunity to reduce the amount of `#ifdef` logic (which Kconfig and devicetree are devoted to avoid) by using a phandle node reference, maybe using the [clocks devicetree API](https://docs.zephyrproject.org/latest/doxygen/html/group__devicetree-clocks.html):

```dts
&rcc {
        compatible = "wch,rcc";
        clocks = <&pll [...]>;
};
```

The Linux devictree might be good for explaining the clock handling on devicetree at a high-level:
https://github.com/devicetree-org/dt-schema/blob/main/dtschema/schemas/clock/clock.yaml

Reply:
In practice: `mul` is a feature of the driver that changes according to the devicetree: good fit for a `const struct wch_rcc_pll { uint32_t mul; }` fed from the "device definition" section frequently present at the end of drivers:

https://github.com/zephyrproject-rtos/zephyr/pull/82704/files#diff-b8e1536825d7a19dc949e8cd98c8434fbab4ad4df8fcccde50d305888c781232R167-R169

Reply:
No clue what you are trying to explain here, could you provide a example?

Reply:
ok nvm figured it out, looking into it.

Reply:
Done.

Reply:
Thank you for digging the meaning and for the modification!